File: Merge Strings Alternately - Leetcode 1768/Merge Strings Alternately - Leetcode 1768.java
public class Solution {
    public String mergeAlternately(String word1, String word2) {
        int A = word1.length(), B = word2.length();
        int a = 0, b = 0;
        StringBuilder result = new StringBuilder();

        while (a < A && b < B) {
            result.append(word1.charAt(a++));
            result.append(word2.charAt(b++));
        }

        while (a < A) {
            result.append(word1.charAt(a++));
        }

        while (b < B) {
            result.append(word2.charAt(b++));
        }

        return result.toString();
    }
}

File: Merge Strings Alternately - Leetcode 1768/Merge Strings Alternately - Leetcode 1768.cpp
#include <string>
using namespace std;

class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int A = word1.size(), B = word2.size();
        int a = 0, b = 0;
        string result;

        while (a < A && b < B) {
            result += word1[a++];
            result += word2[b++];
        }

        while (a < A) {
            result += word1[a++];
        }

        while (b < B) {
            result += word2[b++];
        }

        return result;
    }
};
